### PR TITLE
Point to specific parts of the specification

### LINKED ISSUE #7890: C# spec links in C# reference should point to specific parts of the spec
Many articles in the C# reference contain the following section:

> ## C# Language Specification  
> For more information, see the [C# Language Specification](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/index). The language specification is the definitive source for C# syntax and usage.

While that link is useful, even more useful would be a link to the specific part of the specification that talks about the same topic as the article (e.g. [the sizeof article](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/sizeof) would link to [*Unsafe code*, section *The sizeof operator*](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/unsafe-code#the-sizeof-operator)), especially since the spec is large and the location of the right section might not be obvious (e.g. the `sizeof` operator is not specified along with most operators in *Expressions*, but is instead in *Unsafe Code*).

I suspect that this wasn't done in the past, because the spec used to be a Word file. But now that it's a subsection of docs.MS, linking to a specific part is easy.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: a60b6e17-36cf-bd6c-1b76-d38f1760a816
* Version Independent ID: 0fdeaa81-c3c2-c290-224e-792fcee9b297
* Content: [C# Reference](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/index)
* Content Source: [docs/csharp/language-reference/index.md](https://github.com/dotnet/docs/blob/master/docs/csharp/language-reference/index.md)
* Product: **dotnet-csharp**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

### PR DESCRIPTION
**Pull request by ErlendLandro**
Fixes #7890

I'm not 100% sure the links should have the `.md` file ending, but it looked like it.

### CODE REVIEW

File: docs/csharp/programming-guide/generics/index.md [hunk 1]
```diff
@@ -51,7 +51,7 @@ Generics were added to version 2.0 of the C# language and the common language ru
 -   [Generics in the Run Time](../../../csharp/programming-guide/generics/generics-in-the-run-time.md)  
   
 ## C# Language Specification  
- For more information, see the [C# Language Specification](../../../csharp/language-reference/language-specification/index.md).  
+ For more information, see the [C# Language Specification](../../../csharp/language-reference/language-specification/classes.md).
```
Comment (BillWagner) [inline docs/csharp/programming-guide/generics/index.md]: I'd change this link to `../../../csharp/language-reference/language-specification/types#constructed-types`

That is where the spec references generic definitions and instantiations.

**Review by BillWagner — CHANGES_REQUESTED** [review]
Thanks for making these changes @Mackiovello We really appreciate it.

I also like that you've updated the one paragraph where the text taken from the spec has changed. That was great to spot.

I have one suggestion for a better link destination. Once you make that change, I'll :shipit:

Thanks again.

**Comment by BillWagner** [comment]
@Mackiovello You're using the correct syntax, but there's some build magic going on underneath.

The C# (and VB) language specs are from a dependent repo (see [here](https://github.com/dotnet/docs/blob/master/docfx.json#L28).  That means the links need to point to the MD files from the source.  For example:

```markdown
../../../csharp/language-reference/language-specification/arrays.md#array-initializers
```

Should be:

```markdown
../../../../_csharplang/spec/arrays.md#array-initializers
```

Note the extra `..` in the path.

I'll open a PR in the style guide to make a note of it.

**Comment by Mackiovello** [comment]
Thanks for the clear instructions. It seems to work.

**Review by BillWagner — APPROVED** [review]

**Comment by BillWagner** [comment]
Thanks for making the additional changes @Mackiovello I'll :shipit: now.  

You should see these updates on the live site in the next day or two. 💯

**Pull request merged**